File: interviewbit-hashing-diffk-ii.py
class Solution:
    # @param A : tuple of integers
    # @param B : integer
    # @return an integer
    def diffPossible(self, A, B):
        if len(A) < 2:
            return 0

        nums = {A[0]: 1}
        for num in A[1:]:
            if num - B in nums or num + B in nums:
                return 1
            else:
                nums[num] = 1

        return 0

File: interviewbit-hashing-diffk-ii.java
public class Solution {
    public int diffPossible(final List<Integer> a, int b) {
        if (a.size() >= 2) {
            HashMap<Integer, Integer> vals = new HashMap<Integer, Integer>();
            for (Integer i : a) {
                if (vals.containsKey(b + i) || vals.containsKey(i - b)) {
                    return 1;
                }
                vals.put(i, 1);
            }
        }
        return 0;
    }
}
